File: team1api/src/main/java/com/Team1_Back/constants/ReportTypes.java
package com.Team1_Back.constants;

import java.util.List;
import lombok.extern.slf4j.Slf4j;

@Slf4j
public class ReportTypes {

    public static final String FMT_PDF = "PDF";
    public static final String FMT_EXCEL = "EXCEL";

    public static final String EXPENSE_APPROVED_SUMMARY_PDF = "EXPENSE_APPROVED_SUMMARY_PDF";
    public static final String EXPENSE_APPROVED_SUMMARY_EXCEL = "EXPENSE_APPROVED_SUMMARY_EXCEL";


    // EMPLOYEE(USER)
    public static final String PERSONAL_DETAIL_EXCEL = "PERSONAL_DETAIL_EXCEL";
    public static final String PERSONAL_SUMMARY_PDF  = "PERSONAL_SUMMARY_PDF";

    // ADMIN
    public static final String DEPT_DETAIL_EXCEL     = "DEPT_DETAIL_EXCEL";
    public static final String DEPT_SUMMARY_PDF      = "DEPT_SUMMARY_PDF";
    public static final String AI_STRATEGY_PDF       = "AI_STRATEGY_PDF";

    public record TypeDef(String id, String label, String format, boolean adminOnly) {}

    public static final List<TypeDef> ALL = List.of(
            new TypeDef(PERSONAL_DETAIL_EXCEL, "Personal Detailed Records (Excel)", FMT_EXCEL, false),
            new TypeDef(PERSONAL_SUMMARY_PDF,  "Personal Summary Report (PDF)",     FMT_PDF,   false),

            new TypeDef(DEPT_DETAIL_EXCEL,     "Department Detailed Records (Excel)", FMT_EXCEL, true),
            new TypeDef(DEPT_SUMMARY_PDF,      "Department Summary Report (PDF)",     FMT_PDF,   true),
            new TypeDef(AI_STRATEGY_PDF,       "AI Strategy Insight Report (PDF)",    FMT_PDF,   true),

            new TypeDef(
                    EXPENSE_APPROVED_SUMMARY_PDF,
                    "Approved Expense Summary (PDF)",
                    FMT_PDF,
                    false
            ),
            new TypeDef(
                    EXPENSE_APPROVED_SUMMARY_EXCEL,
                    "Approved Expense Summary (Excel)",
                    FMT_EXCEL,
                    false
            )
    );

    public static TypeDef find(String id) {
        log.warn("[TYPES] find called with='{}'", id);

        return ALL.stream().filter(t -> t.id().equals(id)).findFirst().orElse(null);
    }
}

File: team1/src/constants/reportTypes.js
export const REPORT_TYPES = [
    {
        id: "EXPENSE_APPROVED_SUMMARY_PDF",
        label: "Approved Expense Summary (PDF)",
        outputFormat: "PDF",
        roles: ["EMPLOYEE", "ADMIN"],
    },
    {
        id: "EXPENSE_APPROVED_SUMMARY_EXCEL",
        label: "Approved Expense Summary (Excel)",
        outputFormat: "EXCEL",
        roles: ["EMPLOYEE", "ADMIN"],
    },

    // ✅ ADMIN 전용(원하면 유지)
    // { id: "DEPT_DETAIL_EXCEL", label: "Department Detailed Records (Excel)", outputFormat: "EXCEL", roles: ["ADMIN"] },
    // { id: "DEPT_SUMMARY_PDF", label: "Department Summary Report (PDF)", outputFormat: "PDF", roles: ["ADMIN"] },
    { id: "AI_STRATEGY_PDF", label: "AI Strategy Insight Report (PDF)", outputFormat: "PDF", roles: ["ADMIN"] },
];
